dist/layers.go

```diff
@@ -109,3 +113,44 @@ func embedBuildpackTar(tw *tar.Writer, uid, gid int, bp Buildpack, baseTarDir st
 
 	return nil
 }
+
+func LayerHashes(layerTarPath string) (diffID v1.Hash, digest v1.Hash, err error) {
+	fh, err := os.Open(layerTarPath)
+	if err != nil {
+		return v1.Hash{}, v1.Hash{}, errors.Wrap(err, "opening tar file")
+	}
+	defer fh.Close()
+
+	// h <----------------- +
+	//                      |
+	// zh <- zw (gunzip) <- + <-- fh
+
+	h := sha256.New()
+	zh := sha256.New()
+
+	zw, err := gzip.NewWriterLevel(zh, gzip.DefaultCompression)
```

Comment:
Will `gzip.DefaultCompression` get the correct digest when we are working with remote images given that ggcr uses `gzip.BestSpeed`?

Reply:
Yeah... just looked around `imgutil` and found that we are [using the `tarball.Layer`](https://github.com/buildpack/imgutil/blob/master/remote/remote.go#L308) which would give us a different digest based on what we've found. The problem from what I've found is that it all depends on the client that compresses and pushes the image to the registry (`pack build --publish` vs `docker push` vs any other tool). This means that relying on this information might be a bit too brittle for our use case.

FWIW: I've got a [PR up on GGRC](https://github.com/google/go-containerregistry/pull/574) which would allow us to make it consistent with the `docker` client.

Reply:
FYI: created issue in imgutil (https://github.com/buildpack/imgutil/issues/15) to track necessary compression change.